diff --git a/include/gemmini_params.h b/include/gemmini_params.h
index 768ebf2..f2bfbcf 100644
--- a/include/gemmini_params.h
+++ b/include/gemmini_params.h
@@ -5,11 +5,11 @@
 #include <limits.h>
 
 #define XCUSTOM_ACC 3
-#define DIM 16
+#define DIM 4
 #define ADDR_LEN 32
 #define BANK_NUM 4
-#define BANK_ROWS 4096
-#define ACC_ROWS 1024
+#define BANK_ROWS 16384
+#define ACC_ROWS 4096
 #define MAX_BYTES 64
 #define MAX_BLOCK_LEN (MAX_BYTES/(DIM*1))
 #define MAX_BLOCK_LEN_ACC (MAX_BYTES/(DIM*4))
@@ -78,6 +78,7 @@ typedef uint32_t acc_scale_t_bits;
 #define ACC_SCALE_SIG_BITS 24
 
 #define ACC_READ_SMALL_WIDTH
+#define ACC_READ_FULL_WIDTH
 
 #define HAS_FIRST_LAYER_OPTIMIZATIONS
 
