--- wprobe_acc24_3p33_l3w3.prj/run.tcl.emitted	2026-10-02 04:23:36.032516701 -0400
+++ wprobe_acc24_3p33_l3w3.prj/run.tcl	2026-10-02 04:23:36.036516597 -0400
@@ -28,6 +28,9 @@
 go compile
 
 solution library add ccs_sample_mem
+go architect
+cycle set /add_0/run/while -equal 3  ;# [hand-patch: declared latency]
+cycle set {/add_0/run/v8.write:asn(v8)} -equal 3  ;# [hand-patch: declared latency]
 go assembly
 go extract
 
